composables/useSearch.ts

```diff
@@ -0,0 +1,17 @@
+export default function () {
+  function removeDiacritics(str: string) {
+    return str
+      .normalize('NFD')
+      .replace(/[\u0300-\u036F]/g, '')
+  }
+
+  function searchText(currentItem: string, queryText: string) {
+    const item = removeDiacritics(currentItem.toLowerCase())
+    const query = removeDiacritics(queryText.toLowerCase())
+    return item.includes(query)
+  }
```

Comment:
Why does not use again `localeCompare` rather unicode edit? It looks to match the the job.
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator#options

```
"e".localeCompare("É", "fr", {sensitivity: "base" })
-> 0
```

Also with `usage: search` option.

Reply:
I didn't figure we had the `search` option !
It though it would be usefull only on sorting element.
I will try to make it work with `localeCompare` then :+1:

Reply:
I'm not sure this is what we need: "For filtering a list of strings by testing each list item for a full-string match against a key." So if 0 is returned, it means the strings à matching (with diacritics support) and if it's non-zero we have no match.

But we are doing partial-string matching in our case, that's why we are using `includes`.

Reply:
An alternative to my Diacritics function would be something like this:
```
let items = ['Offenbach', 'Österreich', 'Aardvark', 'Kangaroo'];
let collator = new Intl.Collator('fr', {usage: 'search', sensitivity: 'base'});
let search = 'o';
let matches = items.filter(v => collator.includes(v, search));
```